2226/2/versus.rb

```diff
@@ -0,0 +1,199 @@
+    выебон выёбывается глупизди доебался ебанёшься ебанул ебашит ёбнул ебало
+    ебанулся ебнул жидоёб жидоёбский заёб изъебнулся заебал заебись козлоёб
+    козлоёбище хуй пизда пиздец \* сука
+  ].freeze
+
+  PRONOUNS_ARRAY = %w[
+    ты Ты и И я Я в не Не что на но с как это а А Но У он Он - – Так У у За за Эй
+    то по бы тебе мой же его мы от ведь Ведь где Где про к В до чем еще о без Мы —  
+  ].freeze
+
+  PRONOUNS = PRONOUNS_ARRAY.join('|')
+
+  SWEAR_WORDS = SWEAR_WORDS_ARRAY.join('|')
+
+  def all_data_storage
+    Dir.glob('rap-battles/*').each_with_object({}) do |file_name, hash|
+      hash[file_name] = File.read(file_name)
+    end
+  end
+
+  def list_all_rapers
+    filter_rapers.flatten!.uniq!
+  end
+
+  # :reek:DuplicateMethodCall
+  def number_of_battles(name)
+    battles_storage = []
+
+    all_data_storage.each do |file|
+      battles_storage << file.first.match(name)
+      battles_storage.reject!(&:nil?)
+    end
+    battles_storage << 'at least one battle' if battles_storage.size.zero?
+
+    battles_storage.size / 2
+  end
+
+  def number_of_swear_words(name)
+    swear_words_storage = []
+
+    all_data_storage.inject([]) do |_, file|
+      if file.first.match(name)
+        swear_words_storage << file.last.scan(/#{SWEAR_WORDS}/)
+        swear_words_storage.flatten!
+      end
+    end
+    swear_words_storage.size
+  end
+
+  def average_number_swearing_words_in_battle(name)
+    average = number_of_swear_words(name).fdiv(number_of_battles(name))
+    average.round(2)
+  end
+
+  def number_of_words_in_rounds(name)
+    number_of_words_in_rounds = 0
+
+    all_data_storage.inject([]) do |_, file|
+      number_of_words_in_rounds += file.last.split(/\W/).count if file.first.match(name)
+    end
+    number_of_words_in_rounds
+  end
+
+  # :reek:DuplicateMethodCall
+  def number_of_rounds(name)
+    round_words = []
+
+    all_data_storage.inject([]) do |_, file|
+      if file.first.match(name)
+        round_words << file.last.match(/Раунд \w/)
+        round_words.reject!(&:nil?)
+      end
+    end
+    round_words << 'this raper had at least 1 round' if round_words.size.zero?
+
+    round_words.size
+  end
+
+  def average_number_words_in_round(name)
+    number_of_words_in_rounds(name) / number_of_rounds(name)
+  end
+
+  def statistics(number_of_rapers)
+    unsorted_storage = {}
+    list_all_rapers.each do |raper|
+      unsorted_storage[raper] = [
+        number_of_battles(raper), number_of_swear_words(raper),
+        average_number_swearing_words_in_battle(raper), average_number_words_in_round(raper)
+      ]
+    end
+    sorted_storage = sort_and_reverse_hash(unsorted_storage)
+    sorted_storage.first(number_of_rapers.to_i)
+  end
+
+  def make_table(number_of_rapers)
+    filtered_data = statistics(number_of_rapers).map { |elem| [elem[0], elem[1]] }.to_h
+    rows = []
+    filtered_data.each do |k, v|
+      rows << [
+        k.to_s, "#{v[0]} баттлов", "#{v[1]} нецензурных слов",
```

Comment:
```
[
  value1,
  value2,
  ...
]
```
It's much more readable

Reply:
I need to reduce the number of lines in method, so a followed another syntax style - rubocop respond that it ok
```
[value1,
 vlaue2,
...]
```